python/python/bystro/ancestry/define_callset.py

```diff
@@ -0,0 +1,141 @@
+"""Calculate intersection of Illumina and Affymetrix chips."""
+
+import logging
+import re
+
+import pandas as pd
+import tqdm
+from liftover import get_lifter
+
+from bystro.ancestry.asserts import assert_equals, assert_true
+from bystro.ancestry.train import DATA_DIR, INTERMEDIATE_DATA_DIR
+from bystro.ancestry.train_utils import is_autosomal_variant
+
+logger = logging.getLogger(__name__)
+converter = get_lifter("hg19", "hg38")
+
+ILLUMINA_FILEPATH = DATA_DIR / "Human660W-Quad_v1_H.csv"
+AFFYMETRIX_FILEPATH = DATA_DIR / "Axiom_PMRA.na35.annot.csv"
+
+
+def get_watson_crick_complement(base: str) -> str:
+    """Calculate Watson-Crick Complement."""
+    wc_dict = {"A": "T", "T": "A", "G": "C", "C": "G"}
+    return wc_dict[base]
+
+
+def liftover_38_from_37(variant: str) -> str | None:
+    """Liftover a variant to genome build 38 from 37."""
+    chrom, pos, ref, alt = variant.split(":")
+    if "MT" in chrom:
+        return None
+    # liftover doesn't deal gracefully with MT variants, but we don't need them anyway
+    locations = converter[chrom][int(pos)]
+    if locations is None or len(locations) != 1:
+        logger.debug("Variant %s had a non-unique location, couldn't lift over", variant)
+        return None
+    chrom38, pos38, _strand38 = locations[0]
+    variant38 = ":".join([chrom38, str(pos38), ref, alt])
+    assert_true("lifted-over variant starts with 'chr'", variant38.startswith("chr"))
+    return variant38
+
+
+def _load_illumina_df() -> pd.DataFrame:
+    comment_rows = 7
+    columns_to_keep = ["Chr", "MapInfo", "SNP", "RefStrand"]
+    illumina_df = pd.read_csv(ILLUMINA_FILEPATH, skiprows=comment_rows)
+    assert_equals(
+        "Set of genome builds",
+        {37.1},
+        "actual set of genome builds",
+        set(illumina_df.GenomeBuild.dropna()),
+    )
+    return illumina_df[columns_to_keep].dropna()
+
+
+def load_illumina_variants() -> pd.Series:
+    """Load list of variants for illumina chip."""
```

Comment:
I know the chip name is in the file path but I think it might be helpful to put the names of the chips in the doc strings so it gets captured for future reference and in autodocs.